controllers/platform/services/services.go

```diff
@@ -505,3 +576,167 @@ func mergeContainerPreservingEnvVars(dest *corev1.Container, source *corev1.Cont
+	return &eventingv1.Trigger{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-data-index-%s-trigger", platformName, tag),
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: eventingv1.TriggerSpec{
+			Broker: brokerName,
+			Filter: &eventingv1.TriggerFilter{
+				Attributes: eventingv1.TriggerFilterAttributes{
+					"type": eventType,
+				},
+			},
+			Subscriber: duckv1.Destination{
+				Ref: &duckv1.KReference{
+					Name:       serviceName,
+					Namespace:  namespace,
+					APIVersion: "v1",
+					Kind:       "Service",
+				},
+				URI: &apis.URL{
+					Path: path,
+				},
+			},
+		},
+	}
+}
+func (d *DataIndexHandler) GenerateKnativeResources(platform *operatorapi.SonataFlowPlatform, lbl map[string]string) ([]client.Object, error) {
+	broker := d.GetSourceBroker()
+	if broker == nil {
+		return nil, nil // Nothing to do
+	}
+	brokerName := broker.Ref.Name
+	namespace := platform.Namespace
+	serviceName := d.GetServiceName()
+	return []client.Object{
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-error", "ProcessInstanceErrorDataEvent", pathProcesses),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-node", "ProcessInstanceNodeDataEvent", pathProcesses),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-sla", "ProcessInstanceSLADataEvent", pathProcesses),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-state", "ProcessInstanceStateDataEvent", pathProcesses),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-variable", "ProcessInstanceVariableDataEvent", pathProcesses),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "process-definition", "ProcessDefinitionEvent", pathDefinitions),
+		d.newTrigger(lbl, brokerName, namespace, platform.Name, serviceName, "jobs", "JobEvent", pathJobs)}, nil
+}
+
+func (d JobServiceHandler) GetSourceBroker() *duckv1.Destination {
+	if d.platform.Spec.Services.JobService.Source != nil && d.platform.Spec.Services.JobService.Source.Ref != nil {
+		return d.platform.Spec.Services.JobService.Source
+	}
+	return GetPlatformBroker(d.platform)
+}
+
+func (d JobServiceHandler) GetSink() *duckv1.Destination {
+	if d.platform.Spec.Services.JobService.Sink != nil {
+		return d.platform.Spec.Services.JobService.Source
+	}
+	return GetPlatformBroker(d.platform)
+}
+
+func (j *JobServiceHandler) GenerateKnativeResources(platform *operatorapi.SonataFlowPlatform, lbl map[string]string) ([]client.Object, error) {
+	broker := j.GetSourceBroker()
+	sink := j.GetSink()
+	namespace := platform.Namespace
+	resultObjs := []client.Object{}
+
+	if broker != nil {
+		brokerName := broker.Ref.Name
+		jobCreateTrigger := &eventingv1.Trigger{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "jobs-service-create-job-trigger",
```

Comment:
What is the naming strategy for the Jobs Service instance? Maybe we should append that name here.